nodejs/deploy

```diff
@@ -37,7 +37,16 @@ nvm install ${NODE_VERSION}
 rm -f ~/.nvm_bin
 ln -s $NVM_BIN ~/.nvm_bin
 
-if [ -f ${CURRENT_DIR}/package.json ]; then
-	pushd $CURRENT_DIR && npm install --production
-	popd
+
+if [ -f ${CURRENT_DIR}/package.json ] && [ -f "${CURRENT_DIR}/yarn.lock" ]; then
+    echo "yarn.lock detected, using yarn to install node packages"
+    yarn_bin=$CURRENT_DIR/node_modules/yarn/bin/yarn
+    if [ ! -f $yarn_bin ]; then
+        npm install yarn
+    fi
+    pushd $CURRENT_DIR && ${yarn_bin} install --production
```

Comment:
You should move the `pushd $CURRENT_DIR` to before the `if` checking if yarn is already installed, so that npm installs yarn at the right place.